services/st_util/StBootloaderCommunicatorUart.cpp

```diff
@@ -105,7 +106,8 @@ namespace services
 
     void StBootloaderCommunicatorUart::ReadMemory(uint32_t address, infra::ByteRange& data, const infra::Function<void()>& onDone)
     {
-        really_assert(data.size() <= 255);
+        really_assert(data.size() <= 256);
+        really_assert(data.size() > 1);
```

Comment:
I think we also should support reading 1 byte, although that may need implementation with a kind of workaround. Otherwise some poor fellow a few years from now will be searching really hard why his application crashes at inconvenient moments